src/components/ViewPresignedUrl/index.tsx

```diff
@@ -48,124 +51,185 @@ function ViewPresignedUrl({ presingedUrl }: Props) {
-      () => getPreSignedUrlData(presingedUrl),
-      { enabled: isDownloadable }
+  // Return HTML (via iframe) display
+
+  /** **************************** TEMPORARY SOLUTION ****************************
+   *
+   * Only allow S3 presigned url that is allowed to be viewed in this <iframe/> tag.
+   *
+   * See below 'TEMPORARY SOLUTION' block for details. Change to original solution if the fix has been made.
+   *
+   * Current solution: if (HTML_FILETYPE_LIST.includes(filetype) && presingedUrl.startsWith('https://umccr')) {
+   * Original solution: if (HTML_FILETYPE_LIST.includes(filetype)) {
+   *
+   * */
+  if (HTML_FILETYPE_LIST.includes(filetype) && presingedUrl.startsWith('https://umccr')) {
+    return (
+      <div className='w-full h-full'>
+        <iframe className='w-full h-full bg-white' src={presingedUrl} />
+      </div>
     );
+  }
 
-    useEffect(() => {
-      if (isError) {
-        toastShow({
-          severity: 'error',
-          summary: 'Something went wrong!',
-          detail: 'Unable to load presignedUrl content.',
-          life: 3000,
-        });
-      }
-    }, [isError]);
-
-    if (isLoading || isFetching || !data) {
-      return <CircularLoaderWithText text='Fetching Content' />;
+  // Data fetching
+  const { isFetching, isLoading, isError, data } = useQuery(
+    ['getPresignedContent', presingedUrl],
+    () => getPreSignedUrlData(presingedUrl),
+    {}
+  );
+
+  useEffect(() => {
+    if (isError) {
+      toastShow({
+        severity: 'error',
+        summary: 'Something went wrong!',
+        detail: 'Unable to load presignedUrl content.',
+        sticky: true,
+      });
     }
+  }, [isError]);
+  const [isPrettifyPreview, setIsPrettifyPreview] = useState<boolean>(true);
 
-    if (filetype == 'json') {
-      const cssTheme = {
-        main: 'line-height:1.3;color:#a21515;background:#ffffff;overflow:auto;',
-        error: 'line-height:1.3;color:#a21515;background:#ffffff;overflow:auto;',
-        key: 'color:#a21515;',
-        string: 'color:#0551a5;',
-        value: 'color:#0b8658;',
-        boolean: 'color:#0551a5;',
-      };
-
-      // Sanitize if JSON is
-      try {
-        const JSONParse = JSON.parse(data);
-        return (
-          <JSONPretty
-            id='json-pretty'
-            data={JSONParse}
-            theme={cssTheme}
-            style={{
-              borderRadius: '5px',
-              width: '100%',
-              minWidth: '100%',
-            }}
-          />
-        );
-      } catch (err) {
-        return (
-          <div>
-            <p>ERROR</p>
-          </div>
-        );
-      }
-    }
+  if (isLoading || isFetching || !data) {
+    return <CircularLoaderWithText text='Fetching Content' />;
+  }
 
-    if (
-      [...DELIMITER_SEPARATED_VALUE_FILETYPE_LIST, ...PLAIN_FILETYPE_LIST, 'yaml'].includes(
-        filetype
-      )
-    ) {
+  /** **************************** TEMPORARY SOLUTION ****************************
```

Comment:
We could probably could remove this if the new `header` to override response content-type implemented.

Happy to leave it as it is now, as it shouldn't affect performance wise.

Reply:
Sure, sound good. Thanks for the note.